sample/dir.rb

```diff
@@ -1,12 +1,7 @@
 # directory access
 # list all files but .*/*~/*.o
-dirp = Dir.open(".")
-for f in dirp
-  case f
-  when /\A\./, /~\z/, /\.o\z/
-    # do not print
-  else
-    print f, "\n"
+Dir.open(".").each do |file|
+  unless file.match(/\A\./, /~\z/, /\.o\z/)
```

Comment:
`String#match` takes 1 or 2 arguments, and the second is offset.
```suggestion
  unless file.match?(/\A\.|(?:~|\.o)\z/)
```

Reply:
or
```suggestion
  unless file.start_with?('.') or file.end_with?('~', '.o')
```